v4/export/writer_util.go

```diff
@@ -148,8 +152,11 @@ func WriteInsert(tblIR TableDataIR, w io.Writer) error {
 			counter += 1
 
 			if bf.Len() >= lengthLimit {
-				wp.input <- bf.Bytes()
-				bf.Reset()
+				wp.input <- bf
+				bf = pool.Get().(*bytes.Buffer)
+				if bf.Cap() < lengthLimit {
+					bf.Grow(lengthLimit)
```

Comment:
Sounds like this should be `lengthLimit - bf.Cap()` 🤔

Reply:
Yes, It makes sense.